src/ripple/app/tx/impl/InvariantCheck.cpp

```diff
@@ -135,12 +135,16 @@ XRPNotCreated::visitEntry(
 
 bool
 XRPNotCreated::finalize(
-    STTx const&,
+    STTx const& tx,
     TER const,
     XRPAmount const fee,
     ReadView const&,
     beast::Journal const& j)
 {
+    // AMM is created with AMMInstanceCreate, not payment
+    if (tx.getTxnType() == ttAMM_INSTANCE_CREATE)
```

Comment:
I don't understand this. This is a check for new XRP created, right, not account created. I'm not following why this is needed. Can you explain (and add a comment in the code as well)